graphql/src/query.rs

```diff
@@ -474,31 +475,47 @@ impl Mutation {
     }
 
     /// Adds a bookmark to the user's list of bookmarks.
-    async fn add_bookmark(&self, context: &Context<'_>, document_id: Uuid) -> FieldResult<String> {
+    #[graphql(guard = "AuthGuard")]
+    async fn add_bookmark(
+        &self,
+        context: &Context<'_>,
+        document_id: Uuid,
+    ) -> FieldResult<Option<AnnotatedDoc>> {
         let user = context
             .data_opt::<UserInfo>()
             .ok_or_else(|| anyhow::format_err!("User is not signed in"))?;
-        Ok(context
+        let document = context
+            .data::<DataLoader<Database>>()?
+            .load_one(dailp::DocumentId(document_id))
+            .await?;
+        context
             .data::<DataLoader<Database>>()?
             .loader()
             .add_bookmark(document_id, user.id)
-            .await?)
+            .await?;
+        Ok(document)
```

Comment:
I think we probably want to load the document after we change the data, right? I know in this case it won't matter because we fetch `bookmarked_on` lazily, but it just feels backwards, and it isn't a pattern we should replicate elsewhere.

That would just mean flipping the data loader call and the `add_bookmark` call:

```
        context
            .data::<DataLoader<Database>>()?
            .loader()
            .add_bookmark(document_id, user.id)
            .await?;
        Ok(context
            .data::<DataLoader<Database>>()?
            .load_one(dailp::DocumentId(document_id))
            .await?)
```